### DISCUSSION TITLE
database "plausible_db" does not exist

### DISCUSSION THREAD

**Discussion by navjotjsingh** [plausible/analytics#562] [Self-Hosted Support]
I have tried installing Plausible multiple times and each time I am getting the error that 'plausible_db' doesn't exist. I clean up all images/containers before trying again but I keep getting the same error.

Here is the log

> Creating network "hosting_default" with the default driver
> Creating volume "hosting_db-data" with local driver
> Creating volume "hosting_event-data" with local driver
> Creating volume "hosting_geoip" with local driver
> Creating volume "hosting_caddy-certificates" with local driver
> Creating hosting_mail_1                ... done
> Creating hosting_geoip_1               ... done
> Creating hosting_plausible_events_db_1 ... done
> Creating hosting_plausible_db_1        ... done
> Creating hosting_plausible_1           ... done
> Creating caddy-gen                     ... done
> Attaching to hosting_mail_1, hosting_geoip_1, hosting_plausible_events_db_1, hosting_plausible_db_1, hosting_plausible_1, caddy-gen
> geoip_1                | # STATE: Creating configuration file at /etc/GeoIP.conf
> geoip_1                | # STATE: Running geoipupdate
> geoip_1                | # STATE: Sleeping for 168 hours
> mail_1                 |     1 LOG: MAIN
> mail_1                 |     1   exim 4.89 daemon started: pid=1, -q30m, listening for SMTP on port 25 (IPv6 and IPv4)
> plausible_db_1         | The files belonging to this database system will be owned by user "postgres".
> plausible_db_1         | This user must also own the server process.
> plausible_db_1         |
> plausible_db_1         | The database cluster will be initialized with locale "en_US.utf8".
> plausible_db_1         | The default database encoding has accordingly been set to "UTF8".
> plausible_db_1         | The default text search configuration will be set to "english".
> plausible_db_1         |
> plausible_db_1         | Data page checksums are disabled.
> plausible_db_1         |
> plausible_db_1         | fixing permissions on existing directory /var/lib/postgresql/data ... ok
> plausible_db_1         | creating subdirectories ... ok
> plausible_db_1         | selecting dynamic shared memory implementation ... posix
> plausible_db_1         | selecting default max_connections ... 100
> plausible_db_1         | selecting default shared_buffers ... 128MB
> plausible_db_1         | selecting default time zone ... Etc/UTC
> plausible_db_1         | creating configuration files ... ok
> plausible_db_1         | running bootstrap script ... ok
> plausible_db_1         | performing post-bootstrap initialization ... ok
> plausible_events_db_1  | Processing configuration file '/etc/clickhouse-server/config.xml'.
> plausible_events_db_1  | Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
> plausible_events_db_1  | Merging configuration file '/etc/clickhouse-server/config.d/logging.xml'.
> plausible_events_db_1  | Include not found: clickhouse_remote_servers
> plausible_events_db_1  | Include not found: clickhouse_compression
> plausible_events_db_1  | Logging warning to /var/log/clickhouse-server/clickhouse-server.log
> plausible_events_db_1  | Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
> plausible_events_db_1  | Logging warning to console
> plausible_events_db_1  | Include not found: networks
> plausible_events_db_1  | 2021.01.07 15:03:44.935413 [ 1 ] {} <Warning> Access(local directory): File /var/lib/clickhouse/access/users.list doesn't exist
> plausible_events_db_1  | 2021.01.07 15:03:44.935473 [ 1 ] {} <Warning> Access(local directory): Recovering lists in directory /var/lib/clickhouse/access/
> plausible_events_db_1  | 2021.01.07 15:03:45.003815 [ 1 ] {} <Warning> Application: Listen [::]:8123 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 20.12.4.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
> plausible_events_db_1  | 2021.01.07 15:03:45.004126 [ 1 ] {} <Warning> Application: Listen [::]:9000 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 20.12.4.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
> plausible_events_db_1  | 2021.01.07 15:03:45.004299 [ 1 ] {} <Warning> Application: Listen [::]:9009 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 20.12.4.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
> plausible_events_db_1  | 2021.01.07 15:03:45.004466 [ 1 ] {} <Warning> Application: Listen [::]:9004 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 20.12.4.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
> caddy-gen              | 2021/01/07 15:03:46 Generated '/etc/caddy/Caddyfile' from 6 containers
> caddy-gen              | forego     | starting dockergen.1 on port 5000
> caddy-gen              | forego     | starting caddy.1 on port 5100
> caddy-gen              | dockergen.1 | 2021/01/07 15:03:46 Contents of /etc/caddy/Caddyfile did not change. Skipping notification ''
> caddy-gen              | dockergen.1 | 2021/01/07 15:03:46 Watching docker events
> caddy-gen              | dockergen.1 | 2021/01/07 15:03:46 Contents of /etc/caddy/Caddyfile did not change. Skipping notification ''
> caddy-gen              | caddy.1    | {"level":"info","ts":1610031826.783515,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":""}
> caddy-gen              | caddy.1    | {"level":"info","ts":1610031826.789529,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["localhost:2019","[::1]:2019","127.0.0.1:2019"]}
> caddy-gen              | caddy.1    | 2021/01/07 15:03:46 [INFO][cache:0xc000237440] Started certificate maintenance routine
> caddy-gen              | caddy.1    | Successfully started Caddy (pid=30) - Caddy is running in the background
> caddy-gen              | caddy.1    | 2021/01/07 15:03:46 [INFO][stats.domain.com] Obtain certificate; acquiring lock...
> plausible_db_1         | syncing data to disk ... ok
> plausible_db_1         |
> plausible_db_1         | initdb: warning: enabling "trust" authentication for local connections
> plausible_db_1         | You can change this by editing pg_hba.conf or using the option -A, or
> plausible_db_1         | --auth-local and --auth-host, the next time you run initdb.
> plausible_db_1         |
> plausible_db_1         | Success. You can now start the database server using:
> plausible_db_1         |
> plausible_db_1         |     pg_ctl -D /var/lib/postgresql/data -l logfile start
> plausible_db_1         |
> plausible_db_1         | waiting for server to start....2021-01-07 15:03:46.848 UTC [45] LOG:  starting PostgreSQL 12.5 (Debian 12.5-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
> plausible_db_1         | 2021-01-07 15:03:46.849 UTC [45] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
> plausible_db_1         | 2021-01-07 15:03:46.864 UTC [46] LOG:  database system was shut down at 2021-01-07 15:03:45 UTCplausible_db_1         | 2021-01-07 15:03:46.868 UTC [45] LOG:  database system is ready to accept connections
> plausible_db_1         |  done
> plausible_db_1         | server started
> plausible_db_1         |
> plausible_db_1         | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
> plausible_db_1         |
> plausible_db_1         | waiting for server to shut down....2021-01-07 15:03:47.014 UTC [45] LOG:  received fast shutdown request
> plausible_db_1         | 2021-01-07 15:03:47.016 UTC [45] LOG:  aborting any active transactions
> plausible_db_1         | 2021-01-07 15:03:47.020 UTC [45] LOG:  background worker "logical replication launcher" (PID 52) exited with exit code 1
> plausible_db_1         | 2021-01-07 15:03:47.020 UTC [47] LOG:  shutting down
> plausible_db_1         | 2021-01-07 15:03:47.027 UTC [45] LOG:  database system is shut down
> plausible_db_1         |  done
> plausible_db_1         | server stopped
> plausible_db_1         |
> plausible_db_1         | PostgreSQL init process complete; ready for start up.
> plausible_db_1         |
> plausible_db_1         | 2021-01-07 15:03:47.126 UTC [1] LOG:  starting PostgreSQL 12.5 (Debian 12.5-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit
> plausible_db_1         | 2021-01-07 15:03:47.126 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
> plausible_db_1         | 2021-01-07 15:03:47.126 UTC [1] LOG:  listening on IPv6 address "::", port 5432
> plausible_db_1         | 2021-01-07 15:03:47.128 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
> plausible_db_1         | 2021-01-07 15:03:47.139 UTC [64] LOG:  database system was shut down at 2021-01-07 15:03:47 UTCplausible_db_1         | 2021-01-07 15:03:47.142 UTC [1] LOG:  database system is ready to accept connections
> plausible_events_db_1  | Include not found: clickhouse_remote_servers
> plausible_events_db_1  | Include not found: clickhouse_compression
> caddy-gen              | caddy.1    | 2021/01/07 15:03:47 [INFO] acme: Registering account for email@domain.com
> caddy-gen              | caddy.1    | 2021/01/07 15:03:48 [INFO][stats.domain.com] Waiting on rate limiter...
> caddy-gen              | caddy.1    | 2021/01/07 15:03:48 [INFO] [stats.domain.com] acme: Obtaining bundled SAN certificate given a CSR
> caddy-gen              | caddy.1    | 2021/01/07 15:03:48 [INFO] [stats.domain.com] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/9892715999
> caddy-gen              | caddy.1    | 2021/01/07 15:03:48 http: TLS handshake error from 127.0.0.1:39426: EOF
> caddy-gen              | caddy.1    | 2021/01/07 15:03:49 [INFO][stats.domain.com] Served key authentication certificate (TLS-ALPN challenge)
> caddy-gen              | caddy.1    | 2021/01/07 15:03:49 [INFO][stats.domain.com] Served key authentication certificate (TLS-ALPN challenge)
> caddy-gen              | caddy.1    | 2021/01/07 15:03:49 [INFO][stats.domain.com] Served key authentication certificate (TLS-ALPN challenge)
> caddy-gen              | caddy.1    | 2021/01/07 15:03:49 [INFO][stats.domain.com] Served key authentication certificate (TLS-ALPN challenge)
> caddy-gen              | caddy.1    | 2021/01/07 15:03:54 [INFO] [stats.domain.com] The server validated our request
> caddy-gen              | caddy.1    | 2021/01/07 15:03:55 [INFO] [stats.domain.com] Server responded with a certificate.
> caddy-gen              | caddy.1    | 2021/01/07 15:03:55 [INFO][stats.domain.com] Certificate obtained successfully
> plausible_1            | Loading plausible..
> plausible_1            | Starting dependencies..
> plausible_1            | Starting repos..
> plausible_db_1         | 2021-01-07 15:03:57.984 UTC [71] FATAL:  database "plausible_db" does not exist
> plausible_db_1         | 2021-01-07 15:03:57.986 UTC [72] FATAL:  database "plausible_db" does not exist
> plausible_1            | create Plausible.Repo database if it doesn't exist
> plausible_1            | 15:03:58.016 [error] Postgrex.Protocol (#PID<0.238.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "plausible_db" does not exist
> plausible_1            | 15:03:58.016 [error] Postgrex.Protocol (#PID<0.237.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "plausible_db" does not exist
> plausible_events_db_1  | 2021.01.07 15:03:58.098234 [ 64 ] {} <Error> DynamicQueryHandler: Code: 81, e.displayText() = DB::Exception: Database plausible_events_db doesn't exist, Stack trace (when copying this message, always include the lines below):
> plausible_events_db_1  |
> plausible_events_db_1  | 0. DB::DatabaseCatalog::assertDatabaseExistsUnlocked(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const @ 0xd916bd1 in /usr/bin/clickhouse
> plausible_events_db_1  | 1. DB::DatabaseCatalog::assertDatabaseExists(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const @ 0xd916a62 in /usr/bin/clickhouse
> plausible_events_db_1  | 2. DB::Context::setCurrentDatabase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0xd8998bb in /usr/bin/clickhouse
> plausible_events_db_1  | 3. DB::HTTPHandler::processQuery(DB::Context&, Poco::Net::HTTPServerRequest&, HTMLForm&, Poco::Net::HTTPServerResponse&, DB::HTTPHandler::Output&, std::__1::optional<DB::CurrentThread::QueryScope>&) @ 0xe57a913 in /usr/bin/clickhouse
> plausible_events_db_1  | 4. DB::HTTPHandler::handleRequest(Poco::Net::HTTPServerRequest&, Poco::Net::HTTPServerResponse&) @ 0xe57e565 in /usr/bin/clickhouse
> plausible_events_db_1  | 5. Poco::Net::HTTPServerConnection::run() @ 0x10d4f7d5 in /usr/bin/clickhouse
> plausible_events_db_1  | 6. Poco::Net::TCPServerConnection::start() @ 0x10d85cff in /usr/bin/clickhouse
> plausible_events_db_1  | 7. Poco::Net::TCPServerDispatcher::run() @ 0x10d87711 in /usr/bin/clickhouse
> plausible_events_db_1  | 8. Poco::PooledThread::run() @ 0x10eb4ce9 in /usr/bin/clickhouse
> plausible_events_db_1  | 9. Poco::ThreadImpl::runnableEntry(void*) @ 0x10eb0c7a in /usr/bin/clickhouse
> plausible_events_db_1  | 10. start_thread @ 0x9609 in /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
> plausible_events_db_1  | 11. clone @ 0x122293 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
> plausible_events_db_1  |  (version 20.12.4.5 (official build))
> plausible_events_db_1  | 2021.01.07 15:03:58.098482 [ 65 ] {} <Error> DynamicQueryHandler: Code: 81, e.displayText() = DB::Exception: Database plausible_events_db doesn't exist, Stack trace (when copying this message, always include the lines below):
> plausible_events_db_1  |
> plausible_events_db_1  | 0. DB::DatabaseCatalog::assertDatabaseExistsUnlocked(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const @ 0xd916bd1 in /usr/bin/clickhouse
> plausible_events_db_1  | 1. DB::DatabaseCatalog::assertDatabaseExists(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) const @ 0xd916a62 in /usr/bin/clickhouse
> plausible_events_db_1  | 2. DB::Context::setCurrentDatabase(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0xd8998bb in /usr/bin/clickhouse
> plausible_events_db_1  | 3. DB::HTTPHandler::processQuery(DB::Context&, Poco::Net::HTTPServerRequest&, HTMLForm&, Poco::Net::HTTPServerResponse&, DB::HTTPHandler::Output&, std::__1::optional<DB::CurrentThread::QueryScope>&) @ 0xe57a913 in /usr/bin/clickhouse
> plausible_events_db_1  | 4. DB::HTTPHandler::handleRequest(Poco::Net::HTTPServerRequest&, Poco::Net::HTTPServerResponse&) @ 0xe57e565 in /usr/bin/clickhouse
> plausible_events_db_1  | 5. Poco::Net::HTTPServerConnection::run() @ 0x10d4f7d5 in /usr/bin/clickhouse
> plausible_events_db_1  | 6. Poco::Net::TCPServerConnection::start() @ 0x10d85cff in /usr/bin/clickhouse
> plausible_events_db_1  | 7. Poco::Net::TCPServerDispatcher::run() @ 0x10d87711 in /usr/bin/clickhouse
> plausible_events_db_1  | 8. Poco::PooledThread::run() @ 0x10eb4ce9 in /usr/bin/clickhouse
> plausible_events_db_1  | 9. Poco::ThreadImpl::runnableEntry(void*) @ 0x10eb0c7a in /usr/bin/clickhouse
> plausible_events_db_1  | 10. start_thread @ 0x9609 in /usr/lib/x86_64-linux-gnu/libpthread-2.31.so
> plausible_events_db_1  | 11. clone @ 0x122293 in /usr/lib/x86_64-linux-gnu/libc-2.31.so
> plausible_events_db_1  |  (version 20.12.4.5 (official build))
> plausible_1            | 15:03:58.112 [error] Clickhousex.Protocol (#PID<0.244.0>) failed to connect: ** (ErlangError) Erlang error: "Code: 81, e.displayText() = DB::Exception: Database plausible_events_db doesn't exist (version 20.12.4.5 (official build))\n"
> plausible_1            | 15:03:58.112 [error] Clickhousex.Protocol (#PID<0.245.0>) failed to connect: ** (ErlangError) Erlang error: "Code: 81, e.displayText() = DB::Exception: Database plausible_events_db doesn't exist (version 20.12.4.5 (official build))\n"
> plausible_1            | create Plausible.ClickhouseRepo database if it doesn't exist
> plausible_1            | Creation of Db successful!
> plausible_1            | Loading plausible..
> plausible_1            | Starting dependencies..
> plausible_1            | Starting repos..
> plausible_1            | Running migrations for Elixir.Plausible.Repo
> plausible_1            | Running migrations for Elixir.Plausible.ClickhouseRepo
> plausible_1            | Migrations successful!
> plausible_1            | Loading plausible..
> plausible_1            | Starting dependencies..
> plausible_1            | Starting repos..
> plausible_1            | Admin user created successful!

The site does get created. But I get this as the code snippet

`<script async defer data-domain="domain.com" src="http://localhost:8000/js/plausible.js"></script>`

And after fixing the src, when I use it, plausible doesn't collect any data either.

**Comment by FV-Valraiso** [comment]
Maybe a little bit late.
Did you configure BASE_URL in your plausible-conf.env with the scheme ? https:// or http://

**Comment by Taubin** [comment]
Hopefully it's okay to just reply to this instead of creating another issue for the same errors. 

I am receiving the same: 

```
stats                  | Processing configuration file '/etc/clickhouse-server/config.xml'.
stats                  | Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
stats                  | Merging configuration file '/etc/clickhouse-server/config.d/logging.xml'.
stats                  | Logging warning to /var/log/clickhouse-server/clickhouse-server.log
stats                  | Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
stats                  | Logging warning to console
stats                  | 2021.03.07 18:44:34.987312 [ 46 ] {} <Warning> Application: Listen [::]:8123 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
stats                  | 2021.03.07 18:44:34.990319 [ 46 ] {} <Warning> Application: Listen [::]:9000 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
stats                  | 2021.03.07 18:44:34.990856 [ 46 ] {} <Warning> Application: Listen [::]:9009 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
stats                  | 2021.03.07 18:44:34.991513 [ 46 ] {} <Warning> Application: Listen [::]:9004 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
```

```
ADMIN_USER_EMAIL=my_email
ADMIN_USER_NAME=my_username
ADMIN_USER_PWD=my_epic_secure_password
BASE_URL=http://stats.mydomain.tld
SECRET_KEY_BASE=super_cool_base
```

I believe I have it all correct, but am still receiving the same errors.

**Reply by FV-Valraiso** [reply]
BASE_URL=http://stats.mydomain.tld
should be
BASE_URL=stats.mydomain.tld
I think

**Reply by bensulli** [reply]
I tried this but got this error on launch:
plausible_1            | ERROR! Config provider Config.Reader failed with:
plausible_1            | ** (RuntimeError) BASE_URL must start with `http` or `https`. Currently configured as `analytics.skyraker.ca`

**Reply by FV-Valraiso** [reply]
On my installed version now, I had to put the schemes and it works fine :
BASE_URL=https://stats.mydomain.tld

**Reply by bensulli** [reply]
Thanks @FV-Valraiso, I think my issue differs from the OP's, so I'll retract my comments.
